### PR TITLE
feat(DTFS2-5961): Portal facility exporter column and sorting

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made 
* Added exporter name column to facilities table
* Added ability to sort by exporter
* Added ability to filter by exporter name
* Added e2e tests

### CODE REVIEW

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js [hunk 1]
```diff
@@ -16,17 +16,30 @@ const dashboardFacilitiesFiltersQuery = (
...
   if (!isSuperUser(user)) {
     query.$and = [
       { 'deal.bank.id': user.bank.id },
     ];
   }
 
-  if (filters.length) {
+  const filtered = [];
+  // removes _crsf from dashboardFilters
+  /* eslint-disable no-unused-vars */
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/facilities/facilities-filters-query.js:30]: Will need to review explicit disablement of linting rules.

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk [hunk 2]
```diff
@@ -4,135 +4,171 @@
...
-        text: "Bank’s facility stage",
-        attributes: {
-          "data-cy": "facility__bankStage--header"
+      {% set linkHtml %}
+        {% if not isChecker %}
+          <a href="{{ url }}" class="govuk-link" data-cy="facility__name--link--{{ facility._id }}">
+            {{ facility.name if facility.name else 'Not entered'}}
+          </a>
+        {% else %}
+          <span data-cy="facility__name--text--{{ facility._id }}"> {{ facility.name }} </span>
```
Comment (abhi-markan) [inline portal/templates/dashboard/_macros/dashboard-facilities-table.njk:30]: Leading and Trailing white spaces are required, is it to satisfy the design?
If yes then please ignore this comment.

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk [hunk 3]
```diff
@@ -4,135 +4,171 @@
...
+        {% endif %}
+      {% endset %}
+
+      {% if facility.hasBeenIssued %}
+        {% set stage = 'Issued' %}
+      {% else %}
+        {% set stage = 'Unissued' %}
+      {% endif %}
+
+      {# {% if facility.currency.id and facility.value %}
```
Comment (abhi-markan) [inline portal/templates/dashboard/_macros/dashboard-facilities-table.njk:40]: Can we please remove commented code.

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk [hunk 4]
```diff
@@ -4,135 +4,171 @@
...
+      {% if activeSortByOrder === 'ascending' %}
+        {% set buttonValue = 'descending' %}
+      {% endif %}
+
+      <button
+        type="submit"
+        name="sortBy"
+        value="{{ buttonValue }}"
+        data-cy="facility__exporter--header-button">
+        Exporter
```
Comment (abhi-markan) [inline portal/templates/dashboard/_macros/dashboard-facilities-table.njk:109]: Is `Exporter` available in `{{ buttonValue }}` variable?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Few comments to address please.

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js [hunk 5]
```diff
@@ -16,17 +16,30 @@ const dashboardFacilitiesFiltersQuery = (
...
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [key, value] of Object.entries(dashboardFilters)) {
+    if (!Object.prototype.hasOwnProperty.call(value, '_csrf')) {
```
Comment (IgnatG) [inline portal/server/controllers/dashboard/facilities/facilities-filters-query.js:33]: can we use `if (!value._csrf)` instead?

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk (hunk 4 above)
Comment (Zainzzkk) [reply portal/templates/dashboard/_macros/dashboard-facilities-table.njk:109]: no, it is only what kind of sort such as ascending or descending

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js (hunk 1 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/facilities/facilities-filters-query.js]: done

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk (hunk 2 above)
Comment (Zainzzkk) [reply portal/templates/dashboard/_macros/dashboard-facilities-table.njk:31]: done, rest is kept constant with rest of file, unless you meant whitespace elsewhere

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk (hunk 3 above)
Comment (Zainzzkk) [reply portal/templates/dashboard/_macros/dashboard-facilities-table.njk]: done

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js (hunk 5 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/facilities/facilities-filters-query.js]: done

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js [hunk 6]
```diff
@@ -16,17 +16,29 @@ const dashboardFacilitiesFiltersQuery = (
...
   if (!isSuperUser(user)) {
     query.$and = [
       { 'deal.bank.id': user.bank.id },
     ];
   }
 
-  if (filters.length) {
+  const filtered = [];
+  // removes _crsf from facilitiesFilter
+  Object.values(dashboardFilters).forEach((value) => {
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/facilities/facilities-filters-query.js:30]: We can simplify this using `filter` method

```
Object.values(dashboardFilters)
  .filter((v) => !v._csrf)
  .map((v) => filtered.push(v));

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Couple of comments to address please.

File: portal/server/controllers/dashboard/facilities/facilities-filters-query.js (hunk 6 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/facilities/facilities-filters-query.js]: done

File: portal/templates/dashboard/_macros/dashboard-facilities-table.njk (hunk 2 above)
Comment (Zainzzkk) [reply portal/templates/dashboard/_macros/dashboard-facilities-table.njk]: done

**Pull request merged**